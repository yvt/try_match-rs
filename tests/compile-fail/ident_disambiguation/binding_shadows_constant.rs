fn main() {
    const foo: i32 = 0;
    try_match::try_match!(42, foo);
}
