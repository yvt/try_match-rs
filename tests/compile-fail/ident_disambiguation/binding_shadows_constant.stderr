error[E0530]: match bindings cannot shadow constants
 --> tests/compile-fail/ident_disambiguation/binding_shadows_constant.rs:3:31
  |
2 |     const foo: i32 = 0;
  |     ------------------- the constant `foo` is defined here
3 |     try_match::try_match!(42, foo);
  |                               ^^^ cannot be named the same as a constant
